src/xma/src/xmaapi/xmares.c

```diff
@@ -519,8 +544,11 @@ static XmaResConfig *xma_shm_open(char *shm_filename, XmaSystemCfg *config)
     /* Ensure other processes will fail to open properly until initialzied */
     fchmod(fd, 0200);
     ret = ftruncate(fd, sizeof(XmaResConfig));
-    if (ret)
+    if (ret) {
+        fchmod(fd, 0666);
```

Comment:
Hi @sarab96 .   We don't want' to change the file permissions at this point.  The file must be initialized by the process in control of this file and the file permission bits can only be set to rw-rw-rw when this has completed successfully.   If there is a problem here, it is fatal and we do not wish other processes to read a file that may be in an inconsistent state (i.e. contain an uninitialized mutex).